### PR TITLE
Fixed single type problem in Python.

### PR DESCRIPTION
**Pull request by cseed**
Intern types rather than recording singletons.
Remove canonical field stuff from LoadVCF (unused), move to LoadGDB.

### CODE REVIEW

File: python/hail/typ.py [hunk 1]
```diff
@@ -91,16 +91,17 @@ def _typecheck(self, annotation):
         return
 
 
-class Singleton(type):
+class Intern(type):
```
Comment (cseed) [inline python/hail/typ.py]: @tpoterba Can you sanity check this?  I don't fully understand this meta class business, and I think you wrote this originally.

File: python/hail/typ.py [hunk 2]
```diff
@@ -878,31 +879,32 @@ def _from_java(cls, jtype):
...
-                      'is.hail.expr.TGenotypeRequired$': (TGenotype, True),
-                      'is.hail.expr.TCallOptional$': (TCall, False),
-                      'is.hail.expr.TCallRequired$': (TCall, True)}
-def __singletons__(scala_type):
-    global __singletons
-    if scala_type not in __singletons:
-        type, required = _singleton_classes[scala_type]
-        __singletons[scala_type] = type(required)
-    return __singletons[scala_type]
+__interns = None
```
Comment (catoverdrive) [inline python/hail/typ.py]: I think that this (`__interns` and `__interns__(scala)`) now duplicates a lot of the work that's being done by the Intern superclass is doing in `__call__`. Would it be worth changing it back to something like:

```
class Type()
def _from_java:
    ...
    if class_name in _intern_classes:
        return __interns__(class_name)[0].__call__(__interns__(class_name)[1])
    ...
...
__interns__ = {scala_object: (python_class, req)}
```
or does that not do what I think it does/is it ok as is?

File: python/hail/typ.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/typ.py]: yeah, this seems fine.

File: python/hail/typ.py (hunk 2 above)
Comment (cseed) [reply python/hail/typ.py]: Good point!  Fixed.  Much cleaner.

**Pull request merged**